packages/Search/src/details/DTK_DetailsHeap.hpp

```diff
@@ -0,0 +1,116 @@
+/****************************************************************************
+ * Copyright (c) 2012-2018 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef DTK_DETAILS_HEAP_HPP
+#define DTK_DETAILS_HEAP_HPP
+
+#include <Kokkos_Macros.hpp>
+
+#include <iterator> // iterator_traits
+#include <utility>  // move
+
+namespace DataTransferKit
+{
+namespace Details
+{
+
+template <typename RandomIterator, typename Compare>
+KOKKOS_INLINE_FUNCTION bool isHeap( RandomIterator first, RandomIterator last,
+                                    Compare comp )
+{
+    using DistanceType =
+        typename std::iterator_traits<RandomIterator>::difference_type;
+    DistanceType len = last - first;
+    for ( DistanceType pos = 0; pos < len; ++pos )
+    {
+        for ( DistanceType child : {2 * pos + 1, 2 * pos + 2} )
+            if ( child < len && comp( *( first + pos ), *( first + child ) ) )
+                return false;
+    }
+    return true;
+}
+
+template <typename RandomIterator, typename DistanceType, typename ValueType,
+          typename Compare>
+KOKKOS_INLINE_FUNCTION void __bubbleUp( RandomIterator first, DistanceType pos,
+                                        DistanceType top, ValueType val,
+                                        Compare comp )
+{
+    DistanceType parent = ( pos - 1 ) / 2;
+    while ( pos > top && comp( *( first + parent ), val ) )
+    {
+        *( first + pos ) = std::move( *( first + parent ) );
```

Comment:
Does `std::move` exist in CUDA?

Reply:
I was trying to avoid triggering the reference back to this PR on the Kokkos side but apparently it did not work :/